### PR TITLE
[jaeger] add query healthcheck

### PR DESCRIPTION
**Pull request by sharkymcdongles**
Signed-off-by: sharkymcdongles <bryanfcarmichael@gmail.com>

#### What this PR does

Adds healthcheck for endpoint-based monitoring like statuscake or pingdom.

#### Which issue this PR fixes

- https://github.com/jaegertracing/jaeger-ui/issues/671

#### Checklist

- [x] [DCO](https://github.com/jaegertracing/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Commits are [GPG signed](https://docs.github.com/en/github/authenticating-to-github/about-commit-signature-verification)
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (`[jaeger]` or `[jaeger-operator]`)

### CODE REVIEW

**Comment by sharkymcdongles** [comment]
If someone could take a look at this I'd be so happy. <3

**Comment by naseemkullah** [comment]
Hey @sharkymcdongles, sorry for the delay!, afaict the health is served at `/` and not `/health` as per mentioned [here](https://www.jaegertracing.io/docs/1.21/deployment/#query-service--ui).  Where is `/health` from?

**Comment by naseemkullah** [comment]
However it is on the [admin port](https://github.com/jaegertracing/helm-charts/blob/a711c3cce1ffb11172c5697909e25ede990948a7/charts/jaeger/templates/query-svc.yaml#L23) which is not in [the ingress](https://github.com/jaegertracing/helm-charts/blob/master/charts/jaeger/templates/query-ing.yaml)

`/` would conflict in the same ingress. Any ideas?

**Comment by sharkymcdongles** [comment]
> However it is on the [admin port](https://github.com/jaegertracing/helm-charts/blob/a711c3cce1ffb11172c5697909e25ede990948a7/charts/jaeger/templates/query-svc.yaml#L23) which is not in [the ingress](https://github.com/jaegertracing/helm-charts/blob/master/charts/jaeger/templates/query-ing.yaml)
> 
> `/` would conflict in the same ingress. Any ideas?
@naseemkullah 

I had a ticket in jaeger project and they told me about the health endpoint. Ideally it would be served via the same port as the UI, but it is not. This is the best way to implement it because this causes the path /health to go to the admin port and as you can see here from my test it does report back exactly what you'd want for a health check:

```
wget jaeger-query:16687/health
Connecting to jaeger-query:16687 (10.7.252.148:16687)
saving to 'health'
health               100% |******************************************************************************************************************************************************************************************|   103  0:00:00 ETA
'health' saved
/alertmanager $ cat health
{"status":"Server available","upSince":"2021-01-22T21:19:20.539238081Z","uptime":"113h20m6.465648599s"}
```

File: charts/jaeger/values.yaml [hunk 1]
```diff
@@ -298,6 +298,7 @@ collector:
       # - secretName: chart-example-tls
       #   hosts:
       #     - chart-example.local
+      # healthCheckEnabled: true
```
Comment (naseemkullah) [inline charts/jaeger/values.yaml:301]:
```suggestion
      health
        exposed: false
```

File: charts/jaeger/templates/query-ing.yaml [hunk 2]
```diff
@@ -23,6 +23,12 @@ spec:
               serviceName: {{ template "jaeger.query.name" $ }}
               servicePort: {{ $servicePort }}
     {{- end -}}
+    {{- if .Values.query.ingress.healthCheckEnabled }}
```
Comment (naseemkullah) [inline charts/jaeger/templates/query-ing.yaml:26]:
```suggestion
    {{- if .Values.query.ingress.health.exposed }}
```

**Review by naseemkullah — CHANGES_REQUESTED** [review]
I see so `/health` is not documented but is a thing? yeah thats great no conflict in that case.

nit about value structure

**Comment by sharkymcdongles** [comment]
@naseemkullah I have merged in your proposed changes. I think the version may need to be incremented when you give final go ahead. Currently it is set to 0.39.8, but I am happy to change to whatever you suggest.

**Comment by naseemkullah** [comment]
> @naseemkullah I have merged in your proposed changes. I think the version may need to be incremented when you give final go ahead. Currently it is set to 0.39.8, but I am happy to change to whatever you suggest.

Thanks @sharkymcdongles, seems like unrelated changes have been added to the PR. Could you fix that? If it proves to be too messy to, feel free to close and re-open a PR and we'll be sure to get it merged!

**Comment by sharkymcdongles** [comment]
@naseemkullah oops my bad. when I tried to resign the commit it must've rebased. Give me 5 min and I will fix.

**Pull request closed without merging**